src/modules/fancyzones/editor/FancyZonesEditor/App.xaml.cs

```diff
@@ -87,7 +89,42 @@ private void OnStartup(object sender, StartupEventArgs e)
             _themeManager = new ThemeManager(this);
 
             FancyZonesEditorIO.ParseCommandLineArguments();
-            FancyZonesEditorIO.ParseDeviceInfoData();
+
+            var parseResult = FancyZonesEditorIO.ParseZoneSettings();
+
+            // 10ms retry loop with 1 second timeout
+            if (!parseResult.Item1)
+            {
+                CancellationTokenSource ts = new CancellationTokenSource();
+                Task t = Task.Run(() =>
+                {
+                    while (!parseResult.Item1 && !ts.IsCancellationRequested)
+                    {
+                        Task.Delay(10).Wait();
+                        parseResult = FancyZonesEditorIO.ParseZoneSettings();
+                    }
+                });
+
+                try
+                {
+                    bool result = t.Wait(1000, ts.Token);
+                    ts.Cancel();
+                }
+                catch (OperationCanceledException)
+                {
+                    ts.Dispose();
+                }
+            }
+
+            // Error message if parsing failed
+            if (!parseResult.Item1)
+            {
+                string message = parseResult.Item2 + Environment.NewLine + Environment.NewLine + FancyZonesEditor.Properties.Resources.Error_Parsing_Zones_Settings_User_Choice;
+                if (MessageBox.Show(message, FancyZonesEditor.Properties.Resources.Error_Parsing_Zones_Settings_Title, MessageBoxButton.YesNo) == MessageBoxResult.No)
+                {
+                    Environment.Exit(0);
```

Comment:
We need to log the error.

Reply:
I've added data saving to the file with a notification message box, as it was done for other errors.

Reply:
OK, let's a comment for future reference when we fully enable logging in the editor.